types/src/transaction/transaction.rs

```diff
@@ -264,6 +232,115 @@ fn is_input_and_output_consistent(inputs: &[AssetTransferInput], outputs: &[Asse
     sum.iter().all(|(_, sum)| sum.is_zero())
 }
 
+fn make_tag(mut item: Vec<u8>, mut result: u128) -> u128 {
```

Comment:
IMO, passing a vector reference and an index instead of a mutable vector will be better.

Reply:
For instance, you can refactor this function like the below.
```rust
fn make_tag(item: &[u8]) -> u128 {
    if item.is_empty() {
        return 0
    }
    fn make_tag_internal(item: &[u8], index: usize, mut result: u128) -> u128 {
        if item.len() == index {
            return result
        }
        // Assume that the mask is stored as little endian
        let head = item[index] as u128;
        result = (result << 8) | head as u128;
        make_tag_internal(item, index + 1, result)
    }
    make_tag_internal(item, 0, 0)
}
```

Reply:
It also seems more elegant because the function's arguments are more intuitive. I replaced it with the code you provided.